lib/ibmiconnector.js

```diff
@@ -681,29 +742,44 @@ class IBMiConnector extends SQLConnector {
    * @param {*} value Column value
    * @returns {*} Model property value
    */
-  fromColumnValue(propertyDef, value) {
-    if (value === null || !propertyDef) {
-      return value;
+  fromColumnValue(prop, val) {
+    if (val == null) {
+      return val;
     }
-    switch (propertyDef.type.name) {
-      case 'Number':
-        return Number(value);
-      case 'String':
-        return String(value);
-      case 'Date':
-        return new Date(value);
-      case 'Boolean':
-        return Boolean(value);
-      case 'GeoPoint':
-      case 'Point':
-      case 'List':
-      case 'Array':
-      case 'Object':
-      case 'JSON':
-        return JSON.parse(value);
-      default:
-        return value;
+    if (prop) {
```

Comment:
Nitpick: have you considered to use reverse logic & early return to make the code easier to read and also avoid so many whitespace changes below?

```js
if (!prop) return val;

switch (prop.type.name) {
  // ...
}
return val;
```